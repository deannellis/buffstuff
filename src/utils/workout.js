import moment from 'moment';

export const getWorkouts = ({ a = 0, b = 0 }) => {
    let workouts = []
    if(a == 0) {
        workouts.push('overhead', 'chinup')
    } else {
        workouts.push('bench', 'row')
    }
    if(b == 0 || b == 2) {
        workouts.push('squat');
    } else {
        workouts.push('deadlift');
    }
    return workouts;
}

export const getDisplayName = name => {
    switch(name) {
        case 'bench':
            return 'Bench Press';
        case 'row':
            return 'Row';
        case 'squat':
            return 'Squat';
        case 'deadlift':
            return 'Deadlift';
        case 'overhead':
            return 'Overhead Press';
        case 'chinup':
            return 'Chin-Ups';
        default:
            return name;
    }
};

export const getWarmupWeights = workWeight => {
    const warmup25 = workWeight * 0.25 > 44 ? workWeight * 0.25 : 44;
    const warmup50 = workWeight * 0.50 > 44 ? workWeight * 0.50 : 44;
    const warmup75 = workWeight * 0.75 > 44 ? workWeight * 0.75 : 44;
    return [ warmup25, warmup50, warmup75 ];
}

export const getWeightDistribution = weight => {
    if(weight === 44) {
        return [ 0, 44, 0]
    }
    const oneSide = (weight - 44) / 2;
    return [ oneSide, 44, oneSide ];
}

export const getMonthWorkouts = workouts => {
    let dates = {};
    workouts.forEach(workout => {
        let month = moment(workout.created).format("MMMM");
        let day = parseInt(moment(workout.created).format("D"));
        if(dates.hasOwnProperty(month)) {
            dates[month].push(day);
        } else {
            dates[month] = [ day ];
        }
    });
    // console.log(dates);
    return dates;
}

export const getWorkoutDays = (start, daysInMonth) => {
    let workoutDays = [];
    let checkEven = false;
    let weekday = start + 1;
    for(let d = 1; d <= daysInMonth; d++) {
        if(checkEven) {
            if(d%2 == 0) workoutDays.push(d)
        } else {
            if(d%2 !== 0) workoutDays.push(d)
        }
        if(weekday == 7){
            weekday = 0;
            checkEven = !checkEven;
        };
        weekday ++;
    }
    return workoutDays;
};
