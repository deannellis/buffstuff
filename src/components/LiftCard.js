import React from 'react';
import { Formik, Form } from 'formik';
import * as Yup from 'yup';

import { getDisplayName, roundWeight } from '../utils/workout';
import Button from './Button';
import NumberInput from './NumberInput';

const LiftCard = ({ lift, weights, formIsOpen, toggleForm, updateWeight }) => {
    return (
        <div className="card lift-card" style={{'minHeight': '30.8rem'}}>
            <div>
                <h2>{getDisplayName(lift)}</h2>
                    <p>Work Weight:</p>
                <div className="lift-card__weight">
                    <p>
                        {roundWeight(weights[lift])}
                        <span>lbs</span>
                    </p>
                </div>
            </div>
            <Button clickHandler={toggleForm}>Edit weight</Button>
            <div className={formIsOpen ? 'card__overlay' : 'card__overlay  card__overlay--hidden' }>
                <UpdateWeightForm 
                    currentWeight={weights[lift]}
                    lift={lift}
                    toggleForm={toggleForm}
                    updateWeight={updateWeight}
                />
            </div>
        </div>
    );
}

LiftCard.defaultProps = {
    lift: 'bench',
    weights: {
        bench: 0,
        row: 0,
        squat: 0,
        deadlift: 0,
        overhead: 0,
    },
    formIsOpen: false,
    toggleForm: () => {console.log('toggle form (default prop)')},
    updateWeight: () => {console.log('update weight (default prop)')},
};
 
export default LiftCard;

const UpdateWeightForm = ({ currentWeight, lift, toggleForm, updateWeight }) => (
    <>
        <Formik
            initialValues={{
                [lift]: currentWeight
            }}
            validationSchema={Yup.object({
                [lift]: Yup.string()
                    .required('Required'),
            })}
            onSubmit={(values, { setSubmitting }) => {
                updateWeight(values);
                toggleForm();
                setSubmitting(false);
            }}
        >
            <Form className="form">
                <p>{`Update ${getDisplayName(lift)} Weight`}</p>
                <NumberInput 
                    label="New Weight"
                    name={lift}
                    type="number"
                    min="0"
                    helperText="Enter amount in pounds"
                />
                <Button variant="primary" type="submit">submit</Button>
                <Button type="button" clickHandler={toggleForm}>cancel</Button>
            </Form>
        </Formik>
    </>
)