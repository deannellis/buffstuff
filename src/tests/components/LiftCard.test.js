import React from "react";
import { shallow } from "enzyme";

import LiftCard from "../../components/LiftCard";
import { weights } from "../fixtures/workout";

test("should render Lift Card without props", () => {
	const wrapper = shallow(<LiftCard />);
	expect(wrapper).toMatchSnapshot();
});

test("should render Lift Card with props", () => {
	const weightsProp = weights;
	const wrapper = shallow(
		<LiftCard lift="row" weights={weightsProp} formIsOpen={true} />
	);
	expect(wrapper).toMatchSnapshot();
});

test("should open update weight form", () => {
	const toggleForm = jest.fn();
	const wrapper = shallow(<LiftCard toggleForm={toggleForm} />);

	wrapper.find("Button").prop("clickHandler")();

	expect(toggleForm).toHaveBeenCalled();
});
