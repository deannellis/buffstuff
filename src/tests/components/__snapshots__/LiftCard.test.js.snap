// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Lift Card with props 1`] = `
<div
  className="card lift-card"
  style={
    Object {
      "minHeight": "30.8rem",
    }
  }
>
  <div>
    <h2>
      Row
    </h2>
    <p>
      Work Weight:
    </p>
    <div
      className="lift-card__weight"
    >
      <p>
        115
        <span>
          lbs
        </span>
      </p>
    </div>
  </div>
  <Button
    clickHandler={[Function]}
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Edit weight
  </Button>
  <div
    className="card__overlay"
  >
    <_default
      currentWeight={115}
      lift="row"
      toggleForm={[Function]}
      updateWeight={[Function]}
    />
  </div>
</div>
`;

exports[`should render Lift Card without props 1`] = `
<div
  className="card lift-card"
  style={
    Object {
      "minHeight": "30.8rem",
    }
  }
>
  <div>
    <h2>
      Bench Press
    </h2>
    <p>
      Work Weight:
    </p>
    <div
      className="lift-card__weight"
    >
      <p>
        0
        <span>
          lbs
        </span>
      </p>
    </div>
  </div>
  <Button
    clickHandler={[Function]}
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Edit weight
  </Button>
  <div
    className="card__overlay  card__overlay--hidden"
  >
    <_default
      currentWeight={0}
      lift="bench"
      toggleForm={[Function]}
      updateWeight={[Function]}
    />
  </div>
</div>
`;
